code/controllers/CronTaskController.php

```diff
@@ -97,16 +117,43 @@ public function index(SS_HTTPRequest $request) {
 	 * @param CronTask $task
 	 */
 	public function runTask(CronTask $task) {
-		$cron = Cron\CronExpression::factory($task->getSchedule());
-		$isDue = $this->isTaskDue($task, $cron);
-		// Update status of this task prior to execution in case of interruption
-		CronTaskStatus::update_status(get_class($task), $isDue);
-		if($isDue) {
-			$this->output(get_class($task).' will start now.');
-			$task->process();
-		} else {
-			$this->output(get_class($task).' will run at '.$cron->getNextRunDate()->format('Y-m-d H:i:s').'.');
+
+		$canRunTask = true;
+		$enforceSchedule = true;
+		$isDue = true;
+
+		if(method_exists($task, "canRunTask")) {
+			$canRunTask = $task->canRunTask();
+		}
+
+		if(method_exists($task, "enforceSchedule")) {
```

Comment:
Like @tractorcow said, a piece of documentation, or at least a docblock next to the interface method is needed next to `enforceSchedule` and `canRunTask`, otherwise it looks like an arbitrary hook :-)